src/actions/bookshelf/utils/bookshelf-utils.js

```diff
@@ -124,3 +127,63 @@ export function appendWhereClause (queryBuilder, fieldName, value, matchStrategy
       throw new Error(`Unrecognized match strategy "${matchStrategy}"`)
   }
 }
+
+// -----------------------------------------------------------------------------
+// Accepts the "orderBy" API field value to apply the appropriate
+// "queryBuilder.orderBy" logic.
+//
+// The function requires including:
+// joint        - The joint instance
+// queryBuilder - The queryBuilder instance
+// modelName    - The model name of the main resource
+// -----------------------------------------------------------------------------
+// This logic is supports ordering by columns of associations (via dot
+// notation). If the association is not defined on the source model, the orderBy
+// argument is ignored.
+//
+// NOTE: NULLS are always returned last in both ASC and DESC orders.
+// -----------------------------------------------------------------------------
+export function appendOrderByClause (joint, queryBuilder, modelName, fieldValue) {
+  const orderBy = buildOrderBy(fieldValue)
+
+  orderBy.map(orderOpt => {
+    // Support column from association
+    if (orderOpt.col.includes('.')) {
+      const parts = orderOpt.col.split('.')
+      const assocName = parts[0]
+      const colName = parts[1]
+      const assocModelName = (joint.modelNameOfAssoc[modelName]) ? joint.modelNameOfAssoc[modelName][assocName] : null
+
+      if (assocModelName) {
+        // Obtain model config info to build raw query
+        const mainTableName = joint.model[modelName].prototype.tableName
+        const assocTableName = joint.model[assocModelName].prototype.tableName
+        const mainModelConfig = joint.modelConfig.find(it => it.name === modelName)
+        const assocConfig = mainModelConfig.associations[assocName]
+
+        if (!assocConfig) {
+          // Skip is association name does not exist on the model
+          if (debugOrderBy) console.warn(`[JOINT] The orderBy argument "${assocName}.${colName}" is being ignored because the assoctiation "${assocName}" does not exist for model "${modelName}".`)
+          return // eslint-disable-line no-useless-return
+        } else if (assocConfig.type !== 'toOne') {
+          // Skip if association is not "toOne" (i.e. it is a "many" relationsip)
+          if (debugOrderBy) console.warn(`[JOINT] The orderBy argument "${assocName}.${colName}" is being ignored because the assoctiation "${assocName}" is not of type "toOne".`)
+          return // eslint-disable-line no-useless-return
```

Comment:
Suggest throwing errors instead of logging warnings:

1. Library usage is likely controlled by the app developers.
2. Errors make the code behavior more deterministic.
3. Warnings are better suited for handling unpredictable runtime inputs (e.g., user input, external APIs).

This change simplifies error handling and slightly reduces code complexity.

Reply:
This is where I am confused. If I throw, won't it stop execution if I am try/catching when consuming it? I want it to keep going. Perhaps I should be aggregating the errors into an "errors" object and return it? 

We can discuss this together later.

Reply:
Going to continue this discussion on a new thread to include code suggestion.